server/src/main/java/org/opensearch/action/admin/cluster/shards/TransportCatShardsAction.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.shards;
+
+import org.opensearch.action.admin.cluster.state.ClusterStateRequest;
+import org.opensearch.action.admin.cluster.state.ClusterStateResponse;
+import org.opensearch.action.admin.indices.stats.IndicesStatsRequest;
+import org.opensearch.action.admin.indices.stats.IndicesStatsResponse;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.action.support.TimeoutTaskCancellationUtility;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.tasks.CancellableTask;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+/**
+ * Perform cat shards action
+ *
+ * @opensearch.internal
+ */
+public class TransportCatShardsAction extends HandledTransportAction<CatShardsRequest, CatShardsResponse> {
+
+    private final NodeClient client;
+
+    @Inject
+    public TransportCatShardsAction(NodeClient client, TransportService transportService, ActionFilters actionFilters) {
+        super(CatShardsAction.NAME, transportService, actionFilters, CatShardsRequest::new);
+        this.client = client;
+    }
+
+    @Override
+    public void doExecute(Task parentTask, CatShardsRequest shardsRequest, ActionListener<CatShardsResponse> listener) {
+        final ClusterStateRequest clusterStateRequest = new ClusterStateRequest();
+        clusterStateRequest.setShouldCancelOnTimeout(true);
+        clusterStateRequest.local(shardsRequest.getLocal());
+        clusterStateRequest.clusterManagerNodeTimeout(shardsRequest.getClusterManagerNodeTimeout());
+        clusterStateRequest.clear().nodes(true).routingTable(true).indices(shardsRequest.getIndices());
+
+        clusterStateRequest.setParentTask(client.getLocalNodeId(), parentTask.getId());
```

Comment:
Do we need to track with parent task if cancellation parameter is not specified ? Is there an overhead ?

Reply:
We don't need to set any cancellation parameter as task is of cancellable type.

Reply:
OK so the assert is parentTask is instance of Cancellable. Lets add an assert for it.